### PR TITLE
Merging hand-written code into one package

### PR DESCRIPTION
**Pull request by guoshimin**

### CODE REVIEW

**Comment by guoshimin** [comment]
@jkachmar @jonschoning PTAL

**Comment by jkachmar** [comment]
Two comments:

1) How do you feel about taking the restructuring as an opportunity to make the module naming a bit more consistent? For example:
    - `Kubernetes.OpenAPI.ClientHelper` -> `Kubernetes.Client.Helper`
    - `Kubernetes.OpenAPI.Watch.Client` -> `Kubernetes.Client.Watch`
    - `Kubernetes.OpenAPI.KubeConfig` -> `Kubernetes.Client.KubeConfig`

With this naming scheme, all of the hand-written functions are under the `Kubernetes.Client` namespace, and all of the generated functions are under the `Kubernetes.OpenAPI` namespace.

I haven't given this _too_ much thought though, so there might be good reasons not to do it this way.

2) Is there a better name for the `ClientHelper` module?  I frequently see `Helper` or `Util` modules on Hackage and they can feel fairly vague. Perhaps it would be better to have something like: 
    -`Kubernetes.OpenAPI.KubeConfig` -> `Kubernetes.Client.KubeConfig`
    - `Kubernetes.OpenAPI.ClientHelper` -> `Kubernetes.Client.Config`
...where the `KubeConfig` module holds functions/types concerned with Kubernetes `kubeconfig` functionality and the `Config` module holds functions/types concerned with the client configuration (i.e. the certificate and manager stuff that's currently in there).

**Comment by guoshimin** [comment]
Yeah I'll make these changes.

**Comment by guoshimin** [comment]
PTAL

**Comment by guoshimin** [comment]
@brendandburns can you approve? Thanks!

File: kubernetes-openapi-client/README.md [hunk 1]
```diff
@@ -1,8 +1,49 @@
...
 ## Example
+
+```haskell
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Data.Function           ((&))
+import qualified Kubernetes.OpenAPI.API.CoreV1
+import           Kubernetes.OpenAPI.Client       (dispatchMime)
```
Comment (denibertovic) [inline kubernetes-openapi-client/README.md]: @guoshimin I don't think the README example here reflects the lates module name changes.

**Comment by denibertovic** [comment]
@guoshimin @jkachmar The way I usually approach module naming in this situation is to have "Kubernetes.Client.*" but then re-export whatever I need to "Kubernetes.Client" that way a user of the library only needs to do "import Kubernetes.Client" and they're set to go. Hoever, a more experienced user is free to pick and choose and import stuff from "Kubernetes.Client.*" one by one (for whatever reason).

I don't have an opinion on the names in "Kubernetes.Client.*" as long as the re-export in "Kubernetes.Client" give me as a user all I need.

**Comment by denibertovic** [comment]
One more question/suggestion regarding the package names: Isn't it a little too much to name the 2 packages kubernetes-openapi-client-gen and kubernetes-openapi-client? Seems a bit of a mouthful. Wouldn't it make more sense to give them shorter (memorable names) like `kubernetes-core` and `kubernetes-client` (or event better just `kubernetes`? I understand the allure to suffix `-gen` so folks know it's generated code but 2 prominent examples of such code, amazonka and gogol, don't seem to use this scheme and it makes them much more memorable and more intuitive to use.

**Comment by guoshimin** [comment]
@denibertovic Thanks for the feedback! PTAL.

**Comment by denibertovic** [comment]
@guoshimin LGTM now. 

Do you have any feedback on the package names in my second comment (for another PR of course).

**Comment by guoshimin** [comment]
Oops, missed your second comment. @jonschoning first raised the point whether it was wise to take the name `kubernetes` on hackage and I agreed with him, so we made the package name more qualified. It's possible that we have gone too far. Now that I think about it more, it may be better to have just one package that includes both generated and hand-written code. That's how the other languages under kubernetes-client work, and more intuitive for the user. One complication is that we will have one .cabal file for both generated and hand-written code, and we will need to maintain it manually, but I think it's outweighed by the benefits. So @jonschoning @jkachmar what do you guys think of just having one package named `kubernetes-client`?

**Comment by denibertovic** [comment]
@guoshimin why would it not be wise to use the name "kubernetes" if it's the official api bindings? Please point me to a discussion about this so you don't have to repeat yourself.

That said, I'm :-1: on the single package proposal. Having 2 packages seems perfectly reasonable and quite in spirit for these types of things in the ecosystem. 

FWIW, if the "qualified" name must stay I would vote for keeping the `kubernetes-openapi-client-gen` name but renaming `kubernetes-opeapi-client` to just `kubernetes-client`.

**Comment by jkachmar** [comment]
- 👍 for shorter names (e.g. `kubernetes-client` and `kubernetes-client-core`/`kubernetes-client-gen`)
- 👎 for a single package
  - I feel like if the generated stuff is separate, it's a little easier for someone to run the generator on their own (e.g. for a different version of the API) in a forked version

**Comment by jonschoning** [comment]
1. I'm good with `kubernetes-client`

    - Just to add some context, my reason for avoiding a root name like `kubernetes` is that although the code in this repo may be generated from the official swagger openapi spec, the openapi generator itself doesn't know anything about kubernetes at all and is constrained to generate to the lowest common denominator of "forall" OAS specs.  
     Packages like `gogol` and `amazonka` have tailor-made generators, which have the advantage of being able to customize the codegen specifically for their own use case.
     So I was thinking to just allow some room on hackage, if someone in the future decides to implement tailor-made generator for kubernetes and really improve the UX.
     But I can live with `kubernetes-client` which I think is different enough

2. I think I prefer keeping the generated code in a separate package as well, but am not super opinionated about this one.

**Comment by guoshimin** [comment]
Thank you folks for your input. I'll change the package name to `kubernetes-client`/`kubernetes-client-core`.

**Comment by guoshimin** [comment]
PTAL

File: kubernetes-client/package.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+name: kubernetes-client
+version: 0.1.0.0
+description: |
+  This package contains functions for working with kubeconfig files.
```
Comment (denibertovic) [inline kubernetes-client/package.yaml]: This description should get updated.

File: kubernetes/README.md [hunk 3]
```diff
@@ -8,4 +8,4 @@ OpenAPI-Specification: https://github.com/OAI/OpenAPI-Specification/blob/master/
 
 ## Usage
 
-Please refer to the README of the `kubernetes-client-helper` package.
+Please refer to the README of the `kubernetes-openapi-client` package.
```
Comment (denibertovic) [inline kubernetes/README.md]: s/kubernetes-openapi-client/kubernetes-client/

File: kubernetes-client/package.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+dependencies:
+  - base >=4.7 && <5.0
+  - bytestring >=0.10.0 && <0.11
+  - aeson >=1.0 && <2.0
+  - connection
+  - containers
+  - data-default-class
+  - http-client >=0.5 && <0.6
+  - http-client-tls
+  - kubernetes-openapi-client-gen == 0.1.0.0
```
Comment (denibertovic) [inline kubernetes-client/package.yaml]: This should change to kubernetes-client-core right?

File: kubernetes-client/package.yaml (hunk 2 above)
Comment (guoshimin) [reply kubernetes-client/package.yaml]: Good catch. Done. And added a few more fields.

File: kubernetes-client/package.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  - pem
+  - safe-exceptions <0.2
+  - streaming-bytestring >= 0.1.5 && < 0.2.0
+  - text >=0.11 && <1.3
+  - tls
+  - x509
+  - x509-system
+  - x509-store
+  - x509-validation
+executables:
```
[the remark this answers is not in the document]
Comment (guoshimin) [reply kubernetes-client/package.yaml]: Moved to test.

File: kubernetes/README.md (hunk 3 above)
Comment (guoshimin) [reply kubernetes/README.md]: Done.

File: kubernetes-client/package.yaml (hunk 4 above)
Comment (guoshimin) [reply kubernetes-client/package.yaml]: I will update it when I regen the code to use the new package name.

Comment (denibertovic) [reply kubernetes-client/package.yaml]: :+1:

**Comment by denibertovic** [comment]
LGTM. :+1:

**Pull request merged**